packages/core/test/slice-test.js

```diff
@@ -90,6 +90,11 @@ describe('slice', function () {
   })
 
   describe('takeWhile', function () {
+    it('given canonical empty stream, should return canonical empty', () => {
+      const s = takeWhile(_ => true, empty())
```

Comment:
Since we don’t care whether `f` returns `true` or `false`, I think it would be clearer to use `Boolean`:

```ts
takeWhile(Boolean, empty())
```
Since we don’t care whether `f` returns `true` or `false`, I think it would be clearer to use `Boolean`:

```ts
takeWhile(Boolean, empty())
```

Reply:
> we don’t care whether f returns true or false

Good point.  In fact, how about we make it completely arbitrary? I.e., return a random boolean to ensure we _can't_ care about it's value.

Reply:
I think what matters here is about making the test correct without unnecessarily halting the reader to try to figure out what’s going on, or why `f` returns `true` here and `false` in another place. As you have seen in my tests, I explicitly label my fixture setup. E.g.:

```ts
// Fixture setup
const arbitraryPredicate = _ => Math.random() >= 0.5 ? true : false
// Exercise system
const result = takeWhile(arbitraryPredicate, empty())
// Verify outcome
assert(isCanonicalEmpty(result))
```

Then the reader don’t have to examine the predicate or wonder about it. However, in this instance, I think that `Boolean` works just as well without explicit labels. On the other hand, if we make the random function ternary inline, it requires the reader to examine it.

Reply:
I think `arbitraryPredicate` is pretty clear.  However, I wouldn't define it inline in the test--it could come from a test helper module.  I'm ok with `Boolean`, too, but I don't think it expresses the arbitrariness quite as explicitly.

Reply:
I meant inline as `takeWhile(_ => Math.random() >= 0.5 ? true : false, empty())`. Putting it in a helper module makes it less clear to the reader, should he decides to examine why a test is failing. For tests, I am a strong believer in code duplication for the sake of explicitness.

Reply:
I still think `arbitraryPredicate` more clearly expresses the intent, and I don't see test helper modules as a problem.  That said, I also think `Boolean` is fine, and that it's more important to move this PR forward.

Reply:
Switched to `Boolean` in 241c6fe.